assets/questions/aad.js

```diff
@@ -0,0 +1,309 @@
+                  e[be].append(to)
+                  deg[to] = deg[to] + 1
+                  break
+  
+      ans = ""
+  
+      while True :
+          flag = False
+          for key in deg:
+              if deg[key] == 0 :
+                  flag = True
+                  ans = ans + key;
+                  for it in e[key]:
+                      deg[it] = deg[it] - 1
+                  del deg[key]
+          if (not flag):
+              break;
+      
+      if(not (len(deg) == 0)) :
+          return ""
+  
+      return ans
+`
+ 
+  },
+  tasks: [{
+    instructions: [
+      [
+        'Given a dictionary (a list of words in lexicographic order) of all',
+        'words in an unknown/invented language, find the alphabet (an ordered ',
+        'list of characters) of that language. '
+      ].join(''),
+      'Example dictionary:',
+      '[art, rat, cat, car]',
+      'Alphabet is: "atrc"'
+    ],
+    prerequisiteSkills: ['Topology Sorting', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'find_dict',
+    correctnessTests: [{
+      input: [
+        "a",
+        "ab",
+        "b"
+      ],
+      allowedOutputs: ['ab']
+    }, {
+      input: [
+        "aaec",
+        "aaed",
+        "aac",
+        "abe",
+        "abc",
+        "bed",
+        "bea",
+        "bcc",
+        "bcd"
+      ],
+      allowedOutputs: ['ecdab']
+    }, {
+      input: [
+        "egdi",
+        "egdb",
+        "egej",
+        "egei",
+        "ebjj",
+        "ebjg",
+        "ebgh",
+        "ebgj",
+        "igac",
+        "igae",
+        "igia",
+        "igif",
+        "idcf",
+        "idcc",
+        "idh"
+      ],
+      allowedOutputs: ['afchjgdeib']
+    },
+    {
+      input: [
+        "curwfe",
+        "curwfe",
+        "curwst",
+        "curwsw",
+        "curuyc",
+        "curuyf",
+        "curuga",
+        "curugs",
+        "cuxdeh",
+        "cuxdeu",
+        "cuxdbv",
+        "cuxdbn",
+        "cuxolc",
+        "cuxolb",
+        "cuxojd",
+        "cuxojs",
+        "cpcnow",
+        "cpcnob",
+        "cpcnqu",
+        "cpcnqp",
+        "cpctkj",
+        "cpctkd",
+        "cpctfg",
+        "cpctfa",
+        "cpbjeq",
+        "cpbjec",
+        "cpbjz",
+        "cpbhx",
+        "cpbhc",
+        "khjzs",
+        "khjzu",
+        "khjis",
+        "khjiv",
+        "khfax",
+        "khfay",
+        "khftg",
+        "khftq",
+        "kmima",
+        "kmimd",
+        "kmiej",
+        "kmies",
+        "kmrbm",
+        "kmrbr",
+        "kmrus",
+        "kmrut"
+      ],
+      allowedOutputs: ['ljhmezirxygadoqckfsvntwbup']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Next, double-check your code to make sure it handles unexpected cases',
+        ', such as there\'s no solution to the test case or there\'re more ',
```

Comment:
cases. For instance, some test cases have no solution or multiple possible solutions.